services/nexus/nexus.service.js

```diff
@@ -27,20 +27,30 @@ const searchApiSchema = Joi.object({
     .required(),
 }).required()
 
-const resolveApiSchema = Joi.object({
+const nexus3SearchApiSchema = Joi.object({
+  items: Joi.array()
+    .items(
+      Joi.object({
+        // This schema is relaxed similarly to nexux2SearchApiSchema
+        version: Joi.string().required(),
+      })
+    )
+    .required(),
+}).required()
+
+const nexus2ResolveApiSchema = Joi.object({
   data: Joi.object({
     baseVersion: optionalDottedVersionNClausesWithOptionalSuffix,
     version: optionalDottedVersionNClausesWithOptionalSuffix,
   }).required(),
 }).required()
 
-// https://repository.sonatype.org/nexus-restlet1x-plugin/default/docs/path__artifact_maven_resolve.html
-// https://repository.sonatype.org/nexus-indexer-lucene-plugin/default/docs/path__lucene_search.html
 const queryParamSchema = Joi.object({
   server: optionalUrl.required(),
   queryOpt: Joi.string()
-    .regex(/(:(?:q|g|a|v|p|c|cn|sha1|from|count|repositoryId|e|r)=[\w-. ]+)+/i)
+    .regex(/(:[\w.]+=[\w-. ]+)+/i)
     .optional(),
+  nexus3: Joi.equal(''),
```

Comment:
It doesn't seem like Sonatype roll out a new major version of Nexus very often, though wonder if it might be better to use something non-version specific for the param name, perhaps `version` or `nexusVersion` and only accept values like `2` and `3` (for now/forseeable future). That way whenever we change the default version from 2 to 3 we don't need to add/remove query params. 

Not a blocker for me, just food for thought

Reply:
Sounds sensible, will change the parameter name!